sdk/resource/os.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource // import "go.opentelemetry.io/otel/sdk/resource"
+
+import (
+	"context"
+	"runtime"
+	"strings"
+
+	"go.opentelemetry.io/otel/semconv"
+)
+
+type osTypeDetector struct{}
+
+// Detect returns a *Resource that describes the operating system type the
+// service is running on.
+func (osTypeDetector) Detect(ctx context.Context) (*Resource, error) {
+	return NewWithAttributes(
+		semconv.OSTypeKey.String(strings.ToUpper(runtime.GOOS)),
```

Comment:
[Based on the current specification](https://github.com/open-telemetry/opentelemetry-specification/blob/b4ebec7e272fc6b6ad31a47485c91a2f7a2534d9/specification/resource/semantic_conventions/os.md) these need to be lowercase.

Reply:
Oh, good catch! It seems like it just [changed](https://github.com/open-telemetry/opentelemetry-specification/commit/b4ebec7e272fc6b6ad31a47485c91a2f7a2534d9#diff-9ecbdd017d5e18e6b7ccbf1c82530db4fbdc07032d0fbdcc5a13e658bb0dd2c7) last week.

Reply:
Addressed in ac97d34c17109d7813d073e9bf16a2d51e8ea641.

Although the returned string in `runtime.GOOS` is already lowercase, I left an explicit `ToLower` for it be future-proof just in case the current casing changes.